### ISSUE TITLE
Read jdx from memory

### ISSUE THREAD

**Issue by muammar** [nzhagen/jcamp#14]
Is there any way I can just read a jdx from memory instead that having to save to to a file and reload it?

**Comment by nzhagen** [comment]
Muammar,

I suspect that you are looking for the answer to a different question than
the one you've asked, since JDX is a *file format*, and the JCAMP project I
have here on GitHub is basically a parser for the file format. Once you've
parsed the file, you now have the data in memory ... and is that not what
you are looking for?

- Nathan

**Comment by muammar** [comment]
Let me rephrase that. I have a function that queries some jdx from NIST. I am using `requests` to get them. When you use `response.content` you can save that content to file, and then use it with your parser. However, I can also keep it in memory by using `return response.content` in my function. Then, if I try to use your parsers, your functions expect either a `PATH` to a jdx file or a `TextIOWrapper` object.  I am right now trying to recreate a `TextIOWrapper` but it is not getting easy. That's why I wondered -- how to pass the jdx that I have in memory as an argument?

Just for the record, attached there is the `response.content` that I get when I do a query.


[query.txt](https://github.com/nzhagen/jcamp/files/2933741/query.txt)

**Comment by nzhagen** [comment]
Muammar,

Ah, yes, now that makes sense. And it's not hard to modify the existing
code to do what you want. You only need to skip the "JCAMP_reader()"
function. Currently "JCAMP_reader()" function takes a filename, opens the
file, and sends the filehandle to the "jcamp_reader()" parser function. But
if you instead have a set of JCAMP-formatted text lines already assigned to
a variable (i.e. already in memory), then all you need to do is to pass
that variable to jcamp_read(text_lines). Thus, something like

data = jcamp_read(text_lines)

and no need for a call to "JCAMP_reader()". The parser function
"jcamp_read()" can take either a file handle or a list of text lines ---
both will work fine.

I hope that is what you were looking for!

- Nathan

**Comment by muammar** [comment]
Hi @nzhagen. 

Thanks for your reply. I already tried that but it gave back an empty dictionary. However, I understood what the problem was. `requests.get` was getting the file that I posted above with a string encoded as 'utf-8'. What I did was to take the `response` variable I created from `requests.get`  and decode each line in this way:

```
response = requests.get(something)
content = response.content.splitlines()
content = [line.decode("utf-8") for line in content]
data =  jcamp_read(content)
```

And everything works fine.  Please, feel free to close this report. It would nice to have such information somewhere. Thank you for your work with this module. 

Best.

**Comment by nzhagen** [comment]
@muammar, I've added your suggestion to the documentation page.
